### ISSUE TITLE
auto-cluster-rejection

### ISSUE THREAD

**Issue by mobeets** [hennig-lab/cluster_viewer#1]
Hung-Yun: I was thinking about training a neural network or a logistic regressor using our scoring of clusters based on waveforms and ISIs, so it’s more unbiased (at least unbiased in our team). It would require us to pick a couple of sessions, score the cluster’s credibility (1 to 5, for example, where 1 is super unlikely to be a unit and 5 definitely a unit), and then average across N raters. We can then feed the waveform/ISI info as well as averaged score of that cluster to our model. What do you think?

Me:
I think the 1-5 scoring is probably overkill and 0/1 labels might suffice (since ultimately 0/1 is what we are doing manually anyway). This way manually labeling a session creates more training data, without the cognitive overhead of requiring a 1-5 rating
We should consider using an autoencoder as in the paper we just read at the journal club, to reduce the 64D waveform to a lower-dimensional representation. (I wonder if we even use the ISI distribution—for myself it looks like the waveforms are much easier to rule out, but curious your thoughts)
A self-supervised approach will probably also be useful, because while we will have some manually labeled sessions, we definitely have a TON of unlabeled waveforms (aka 24/7 recordings)
In general we'll want to ensure waveforms from new patients doesn't become out-of-distribution. Maybe self-supervised approaches help with this, but not sure
